src/installer/hash.ts

```diff
@@ -59,7 +60,16 @@ type HashFileContents = { algorithm: HashAlgorithm, hash: string, hashFileUrl: s
  * Fetches the hashes for the clientFile's directory, then extracts and returns the hash for the given clientFile.
  */
 async function getOnlineHash(clientFile: ClientFile): Promise<HashFileContents | undefined> {
-    const directoryContents = await getDirContents(clientFile.directoryUrl);
+    let directoryContents;
+
+    // Directory URL will always be missing in the clients installed from Github.
+    // So, considering this as a filtering parameter
```

Comment:
i'd call the "filtering parameter" a "discriminator", ie a key whose value tells the type of the object.

Reply:
Yea Okay